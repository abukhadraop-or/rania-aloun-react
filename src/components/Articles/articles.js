import styled from 'styled-components';

export const Container = styled.div`
  margin: 3rem 0;
  width: 100%;
`;

export const ArticlePreview = styled.div`
  border-top: 0.13rem solid #e6e5e3;
  padding: 1.5rem 0;
`;

export const ArticleTitle = styled.p`
  color: black;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 1rem 0 0;
`;

export const ArticleLink = styled.p`
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.8rem;
  margin: 0.2rem 0 0 0;
`;

export const LastLineContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 1rem 0 0 0;
`;

export const ReadMoreP = styled.p`
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.7rem;
  margin: 1rem 0;
`;

export const ArticleTagsContainer = styled.div`
  align-items: center;
  display: flex;
`;

export const ArticleTag = styled.p`
  border-radius: 0.8rem;
  border: 0.063rem solid #bbbbbb;
  color: #bbbbbb;
  cursor: pointer;
  font-size: 0.7rem;
  height: fit-content;
  margin: 0.125rem;
  padding: 0.2rem;
  width: fit-content;
  &:hover {
    border: 0.063rem solid #6d6f70;
  }
`;
