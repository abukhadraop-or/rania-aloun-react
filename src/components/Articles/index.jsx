import React from 'react';
import ContentNav from 'components/ContentNav';
import DisplayArticles from 'utilities/displayArticles';
import PropType from 'prop-types';
import Article from 'components/Article';

function Articles({ articles, selectedTag, currentPage }) {
  const { data } = DisplayArticles(articles, selectedTag, currentPage);
  return (
    <>
      <ContentNav />
      {data.map((article) => (
        <Article data={article} key={article.id} />
      ))}
    </>
  );
}

Articles.propTypes = {
  articles: PropType.arrayOf(
    PropType.shape({
      id: PropType.number,
    })
  ).isRequired,
  selectedTag: PropType.string.isRequired,
  currentPage: PropType.number.isRequired,
};

export default Articles;
