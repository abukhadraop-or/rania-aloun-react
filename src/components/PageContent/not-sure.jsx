// import { Container } from 'components/Article/article';
// import React from 'react';
// import { render, unmountComponentAtNode } from 'react-dom';
// import { act } from 'react-dom/test-utils';
// import PageContent from '.';

// let container = null;
// beforeEach(() => {
//   // setup a DOM element as a render target
//   container = document.createElement('div');
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   // cleanup on exiting
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });

// it('renders article data', async () => {
//   const fakeArticle = {
//     articleTitle: 'testing',
//     createdAt: '2021-09-27T12:33:54.133Z',
//     id: 10,
//     liked: 27,
//     link: 'https://css-tricks.com/snippets/css/css-box-shadow/',
//     publishDate: '2021-05-08T21:00:00.000Z',
//     tags: [
//       {
//         ArticleId: 1,
//         TagId: 1,
//         createdAt: '2021-09-27T03:27:53.000Z',
//         updatedAt: '2021-09-27T03:27:53.000Z',
//       },
//     ],
//     updatedAt: '2021-10-06T12:06:28.367Z',
//     userName: 'testing',
//   };
//   jest.spyOn(global, 'fetch').mockImplementation(() =>
//     Promise.resolve({
//       json: () => Promise.resolve(fakeArticle),
//     })
//   );

//   expect();
// });
