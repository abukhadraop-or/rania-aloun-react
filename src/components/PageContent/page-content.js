import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  margin: 0 6rem;
  @media (max-width: 375px) {
    display: block;
    margin: 0 0.5rem;
  }
`;

export const InnerContainer = styled.div`
  margin: 3rem 0;
  width: 100%;
`;
