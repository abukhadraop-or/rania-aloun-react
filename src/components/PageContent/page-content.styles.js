import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  margin: 0 6rem;
  @media (max-width: 23.4375rem) {
    display: block;
    margin: 0 0.5rem;
  }
  @media (max-width: 26.875rem) {
    margin: 0 0.5rem;
  }
`;

export const InnerContainer = styled.div`
  margin: 3rem 0;
  width: 100%;
`;
