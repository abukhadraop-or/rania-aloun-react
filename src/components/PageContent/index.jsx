import React, { useEffect, useState } from 'react';
import Tags from 'components/Tags';
import Articles from 'components/Articles';
import Pagination from 'components/Pagination';
import getArticles from 'services/articles';
import DisplayArticles from 'utilities/displayArticles';
import { Container, InnerContainer } from './page-content';

function PageContent() {
  const [articles, setArticles] = useState([]);
  const [currentPage, setCurrentPage] = useState(1);
  const [selectedTag, setSelectedTag] = useState('allArticles');

  async function fetchArticles() {
    const { data } = await getArticles();
    setArticles(data);
  }

  useEffect(() => {
    fetchArticles();
  }, []);

  const handlePageChange = (page) => {
    setCurrentPage(page);
  };

  const handleTagSelect = (tag) => {
    setCurrentPage(1);
    setSelectedTag(tag);
  };
  const { totalCount } = DisplayArticles(articles, selectedTag, currentPage);

  return (
    <Container>
      <Tags onTagSelect={handleTagSelect} />
      <InnerContainer>
        <Articles
          articles={articles}
          selectedTag={selectedTag}
          currentPage={currentPage}
        />
        <Pagination itemsCount={totalCount} onPageChange={handlePageChange} />
      </InnerContainer>
    </Container>
  );
}

export default PageContent;
