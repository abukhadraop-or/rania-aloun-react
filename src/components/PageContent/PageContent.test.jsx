import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { act } from '@testing-library/react';
import getArticles from 'services/articles-service';
import Articles from 'components/Articles';
// import Pagination from 'components/Pagination';
import { toast } from 'react-toastify';

import PageContent from '.';

jest.mock('services/articles-service', () => jest.fn());

const toastErrorSpy = jest.spyOn(toast, 'error');

const mockedData = {
  data: {
    articles: [
      {
        articleTitle: 'The way we look at things',
        createdAt: '2021-09-27T12:33:54.133Z',
        id: 1,
        liked: 32,
        link: 'https://css-tricks.com/snippets/css/css-box-shadow/',
        publishDate: '2021-05-08T21:00:00.000Z',
        updatedAt: '2021-10-12T10:11:00.709Z',
        userName: 'Rania',
        tags: [
          {
            TagArticle: {
              ArticleId: 1,
              TagId: 1,
              createdAt: '2021-09-27T03:27:53.000Z',
              updatedAt: '2021-09-27T03:27:53.000Z',
            },
            createdAt: '2021-09-27T03:27:53.000Z',
            id: 1,
            name: 'Article',
            updatedAt: '2021-10-12T10:11:00.709Z',
          },
        ],
      },
    ],
    currentPage: 0,
    totalItems: 4,
    totalPages: 2,
  },
};

describe('<PageContent />', () => {
  it('matches screenshot', () => {
    const wrapper = shallow(<PageContent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('sets fetched data to articles', async () => {
    const promise = Promise.resolve();
    getArticles.mockReturnValue(mockedData);
    const wrapper = mount(<PageContent />);

    await act(() => promise);
    wrapper.update();

    const { articles } = wrapper.find(Articles).props();
    expect(articles).toEqual(mockedData.data.articles);
  });

  // it('it changes currentPage state on pageHandler called', () => {
  //   const wrapper = shallow(<PageContent />);

  // });

  // it('calls handlePageChange on click on a page number', () => {
  // const wrapper = shallow(<PageContent />);
  // let { onPageChange } = wrapper.find(Pagination).props();
  // let page = 0;
  // const mockedPageHandler = (newval) => {
  //   jest.fn((page = newval));
  // };
  // onPageChange = mockedPageHandler;
  // onPageChange(1);
  // console.log('page new = ', page);
  // onPageChange.simulate('change', )

  // const setState = jest.fn();
  // const useStateSpy = jest.spyOn(React, 'useState');
  // useStateSpy.mockImplementation((init) => [init, setState]);
  // expect(setState).toHaveBeenCalled();
  // });
});
