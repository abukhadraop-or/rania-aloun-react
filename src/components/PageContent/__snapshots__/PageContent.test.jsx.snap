// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContent /> matches screenshot 1`] = `
<styled.div>
  <Tags
    onTagSelect={[Function]}
  />
  <styled.div>
    <Articles
      articles={Array []}
      currentPage={0}
      selectedTag="allArticles"
    />
    <Pagination
      onPageChange={[Function]}
      totalArticles={0}
    />
  </styled.div>
</styled.div>
`;
